integration-cli/docker_cli_push_test.go

```diff
@@ -663,3 +664,55 @@ func (s *DockerRegistriesSuite) TestPushNeedsAuth(c *check.C) {
 		c.Fatalf("Wanted %s, got %s", expected, out)
 	}
 }
+
+func (s *DockerRegistrySuite) TestPushWithSkipSchema2(c *check.C) {
+	c.Assert(s.d.StartWithBusybox("--skip-schema2-push=true"), check.IsNil)
+
+	repo := fmt.Sprintf("%s/runcom/busybox", s.reg.url)
+	out, err := s.d.Cmd("tag", "busybox", repo)
+	c.Assert(err, check.IsNil, check.Commentf(out))
+
+	out, err = s.d.Cmd("push", repo)
+	c.Assert(err, check.IsNil, check.Commentf(out))
+
+	digest1 := reference.DigestRegexp.FindString(out)
+	c.Assert(len(digest1), checker.GreaterThan, 0, check.Commentf("no digest found for pushed manifest"))
+
+	out, err = s.d.Cmd("pull", repo+"@"+digest1)
+	c.Assert(err, check.IsNil, check.Commentf(out))
+
+	out, err = s.d.Cmd("rmi", "-f", repo+"@"+digest1)
+	c.Assert(err, check.IsNil, check.Commentf(out))
+
+	c.Assert(s.d.Restart("--skip-schema2-push=false"), check.IsNil)
```

Comment:
Now there's just a comparison of digests being different for different value of flag, which could be caused by different factors. Based on that one cannot guess whether `--skip-schema2-push=true` caused docker to push v1 or v2.

With a push by digest, you could verify when the conversion from v2 to v1 happens:

```
d1 = pushNormal("repo:tag")
d2 = pull("repo@"+d1)
assertEqual(d1, d2)
// no conversion
assertEqual(pushNormal("repo@"+d1), d1)
// conversion v2->v1 happened
assertNotEqual(pushWithSkipV2("repo@"+d1), d1)
```